Core/src/IfcGeometryConverter/CurveConverter.h

```diff
@@ -1001,73 +1001,75 @@ namespace OpenInfraPlatform {
 					//	END_ENTITY;
 					// **************************************************************************************************************************
 					// determine position
-					carve::math::Matrix conic_position_matrix = placementConverter->convertIfcAxis2Placement(circle->Position);
+					carve::math::Matrix conicPositionMatrix = placementConverter->convertIfcAxis2Placement(circle->Position);
 
 					// Get radius
-					double circle_radius = 0.0;
-					if (circle->Radius) {
-						circle_radius = circle->Radius * UnitConvert()->getLengthInMeterFactor();
-					}
-					else {
-						throw oip::InconsistentGeometryException(circle, "No radius!");
-					}
-
-					carve::geom::vector<3> circle_center =
-						conic_position_matrix * carve::geom::VECTOR(0, 0, 0);
+					double circleRadius = circle->Radius * UnitConvert()->getLengthInMeterFactor();
+					
+					carve::geom::vector<3> circleCenter =
+						conicPositionMatrix * carve::geom::VECTOR(0, 0, 0);
 
 					//Calculate an angle on the circle for trimming begin.
-					// Internal TODO: Implement function GetPointOnCurve, which will be able calculate trimming for each curve
-					double start_angle = calculateTrimmingPointOnCircle(circle, trim1Vec, circle_center, circle_radius);
+					double startAngle = 0.; 
+					for (const auto& element : trim1Vec)
+					{
+						carve::geom::vector<3> point = getPointOnCurve(circle, *element);
+						startAngle = getAngleOnCircle(circleCenter, circleRadius, point);
+					}

```

Comment:
For future reference: here, the `TrimmingPreference` should be accounted for. Not needed right now, but please look into it.

Reply:
Resolved: 1c280cd